internal/extproc/translator/gemini_helper.go

```diff
@@ -258,6 +258,76 @@ func assistantMsgToGeminiParts(msg openai.ChatCompletionAssistantMessageParam) (
 	return parts, knownToolCalls, nil
 }
 
+// openAIToolsToGeminiTools converts OpenAI tools to Gemini tools
+// This function combines all the openai tools into a single Gemini Tool as distinct function declarations.
+// This is mainly done because some Gemini models do not support multiple tools in a single request.
+// This behavior might need to change in future based on model capabilities.
+func openAIToolsToGeminiTools(openaiTools []openai.Tool) ([]genai.Tool, error) {
+	if len(openaiTools) == 0 {
+		return nil, nil
+	}
+	var functionDecls []*genai.FunctionDeclaration
+	for _, tool := range openaiTools {
+		if tool.Type == openai.ToolTypeFunction {
+			if tool.Function != nil {
+				var params map[string]any
+				var err error
+
+				if tool.Function.Parameters != nil {
+					switch paramsRaw := tool.Function.Parameters.(type) {
+					case string:
+						if err = json.Unmarshal([]byte(paramsRaw), &params); err != nil {
```

Comment:
pure question: could you clarify exactly what kind of string can be parsed into `map[string]any` here? I don't see a OK test case on this. could you add one?